share/spack/gitlab/cloud_e4s_pipelines/stacks/data-vis-sdk/spack.yaml

```diff
@@ -0,0 +1,69 @@
+spack:
+  view: false
+  concretization: separately
+
+  config:
+    install_tree:
+      root: /home/software/spack
+      padded_length: 512
+      projections:
+        all: '{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+
+  # Many of these are on by default but we want to be explicit to make sure
+  # we iterate through all the desired combinations.  Duplicate specs will
+  # get filtered out by spack-ci so only one build job for a given spec gets
+  # generated regardless of how many times it appears
+  #
+  # The following variants are currently disabled:
+  #   paraview: too big for the CI cloud at the moment
+  #   ascent: known build failures
+  #   sz: known build failures
+  #
+  specs:
+  - matrix
+    - - ecp-data-vis-sdk ~sz
+      - ecp-data-vis-sdk ~sz +adios2
+      - ecp-data-vis-sdk ~sz +darshan
+      - ecp-data-vis-sdk ~sz +faodel
+      - ecp-data-vis-sdk ~sz +hdf5
+      - ecp-data-vis-sdk ~sz +pnetcdf
+      - ecp-data-vis-sdk ~sz +unifyfs
+      - ecp-data-vis-sdk ~sz +veloc
+      - ecp-data-vis-sdk ~sz +vtkm
+      - ecp-data-vis-sdk ~sz +zfp
+      # - ecp-data-vis-sdk+ascent
+      # - ecp-data-vis-sdk+paraview ^paraview+osmesa
+      # - ecp-data-vis-sdk+sz
+
+  mirrors: { "mirror": "s3://spack-binaries-develop/e4s-new-cluster" }
+
+  gitlab-ci:
+    image: { "name": "ghcr.io/scottwittenburg/ecpe4s-ubuntu18.04-runner-x86_64:2020-09-01", "entrypoint": [""] }
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd share/spack/gitlab/cloud_e4s_pipelines/stacks/data-vis-sdk
+      - spack env activate --without-view .
+      - spack -d ci rebuild
+    mappings:
+      - match: [vtk-h, vtk-m, paraview, llvm, vtk]
+        runner-attributes:
+          tags: ["spack", "public", "xlarge", "x86_64"]
+      - match: ['@:']
+        runner-attributes:
+          tags: ["spack", "public", "large", "x86_64"]
+    temporary-storage-url-prefix: "s3://spack-binaries-prs/pipeline-storage"
+    service-job-attributes:
+      before_script:
+        - . "./share/spack/setup-env.sh"
+        - spack --version
+        - cd share/spack/gitlab/cloud_e4s_pipelines/stacks/data-vis-sdk
+        - spack env activate --without-view .
+      tags: ["spack", "public", "medium", "x86_64"]
```

Comment:
This section does not inherit anything from the root level of the `gitlab-ci` section in the same way the entries in `mappings` do, so `service-job-attributes` needs an image.  This is what is causing the failures of the last two jobs (`cleanup` and `rebuild-index`) with the error:

```
==> Error: SPACK_ROOT must point to spack's prefix when using sh
Run this with the correct prefix before sourcing setup-env.sh:
    export SPACK_ROOT=</path/to/spack>
```